src/Database/LSMTree/Internal/Unsafe.hs

```diff
@@ -231,12 +370,20 @@ data Session m h = Session {
       --
       -- INVARIANT: once the session state is changed from 'SessionOpen' to
       -- 'SessionClosed', it is never changed back to 'SessionOpen' again.
-      sessionState  :: !(RWVar m (SessionState m h))
-    , sessionTracer :: !(Tracer m LSMTreeTrace)
+      sessionState       :: !(RWVar m (SessionState m h))
+    , lsmTreeTracer      :: !(Tracer m LSMTreeTrace)
+    , sessionTracer      :: !(Tracer m SessionTrace)
```

Comment:
Why do we add the `sessionTracer` here, separate from the `LSMTreeTrace` one? Below we don't seem to use it, we pass in a `tr` and locally construct a `sessionTracer`, like
```
  where
    sessionTracer = TraceSession (SessionId dir) `contramap` tr
```